openmcyclus/DepleteReactor.py

```diff
@@ -8,8 +11,603 @@ class DepleteReactor(Facility):
+        then a cycle start is recorded. 
+
+        If it's in the middle of a cycle and the core is full, the 
+        the power_cap value is recorded as power generated. If these 
+        conditions aren't met, then a power of 0 is recorded. 
+
+        If it's in the middle of a cycle or the core is full, then 
+        the cycle duration counter increases by one. 
+        '''
+        print("time:", self.context.time, "tock")
+        if self.retired():
+            return
+        
+        if (self.cycle_step >= self.cycle_time+self.refuel_time) and (self.core.count == self.n_assem_core) and (self.discharged == True):
+            self.discharged = False
+            self.cycle_step = 0
+        
+        if (self.cycle_step == 0) and (self.core.count == self.n_assem_core):
+            #self.record("CYCLE_START", "")
+            print("Cycle start")
+
+        if (self.cycle_step >=0) and (self.cycle_step < self.cycle_time) and (self.core.count == self.n_assem_core):
+            lib.record_time_series(lib.POWER, self, self.power_cap)
+            lib.record_time_series("supplyPOWER", self, int(self.power_cap))
+            print("time:", self.context.time, "record 100 power")
+        else:
+            lib.record_time_series(lib.POWER, self, 0)
+            lib.record_time_series("supplyPOWER", self, int(0))
+            print("time:", self.context.time, "record 0 power")
+
+        if (self.cycle_step > 0) or (self.core.count == self.n_assem_core):
+            self.cycle_step += 1  
+            print("time:", self.context.time, "end tock, cycle step:", self.cycle_step)
+        return 
+
+    def enter_notify(self):
+        super().enter_notify()       
+        if len(self.fuel_prefs) == 0:
+            self.fuel_prefs = [0]*len(self.fuel_incommods)
+  
+    def check_decommission_condition(self):
+        '''
+        If the core and the spent fuel are empty, then the core can be 
+        decommissioned.
+        '''
+        print("time:", self.context.time, "core count:", self.core.count)
+        if (self.core.count == 0) and (self.spent_fuel.count == 0):
+            return True
+        else:
+            return False
+
+    def get_material_requests(self): # phase 1
+        '''
+        Send out bid for fuel_incommods. 
+
+        The number of assemblies to order is the total number needed 
+        for the core less what is currently in the core plus 
+        how many need to be in the fresh fuel storage less how many 
+        exist there already. 
+
+        If the reactor has a finite lifetime, calculate the number of 
+        cycles left and the number of assemblies needed for the 
+        lifetime of the reactor. Order whichever number is 
+        lower. 
+        
+        If the reactor does not need more fuel or is retired, then 
+        submit no bids for materials.
+        '''
+        port = []
+        n_assem_order = self.n_assem_core - self.core.count + self.n_assem_fresh + self.fresh_fuel.count
```

Comment:
Based on the docstring, shouldn't this be `... - self.fresh_fuel.count`?

Reply:
Yes it should be. Thank you for catching that.